bin/ch_misc.c

```diff
@@ -278,6 +278,63 @@ struct env_var env_var_parse(const char *line, const char *path, size_t lineno)
    return (struct env_var){ name, value };
 }
 
+/* Take input string and return the same string with the character        
+   substitutions ':' -> '+' and '/' -> '%'. */
+char *fmt_str(char *str) 
+{
+  char *new_str = replace_char(str, ':', '+');
+  new_str = replace_char(new_str, '/', '%');
+  return new_str;
+}
+
+/* Convert image name to correct path if applicable while also ensuring that 
+   the "name" provided isn't a path in the storage directory. */
+char *get_img_path(char *name, bool unsafe, bool writable, char *storage)
+{
+  char *path = NULL;
+  if(!storage) {storage = get_storage_dir();}
+  if(path_subdir_p(storage, name)) // specified "name" is subdir of storage (bad)
+  {
+   if(unsafe)
+   {
+      return name;
+   } else {
+      FATAL("Specified path is in storage (hint: try running image by name)");
+   }
+  } else if(path_exists(name, NULL, false)) // is 'name' a valid path?
+  {
+   return name;
+  } else { // Assume 'name' is image name, try to find it in storage.
+  T_ (1 <= asprintf(&path, "%s/%s", storage, name));
+   if(!path_exists(path, NULL, false)) // make sure the path we constructed is there
+   {
+      if(path_exists(storage, NULL, false)){
+         FATAL("%s not found in storage", name);
+      } else { // Storage dir doesn't exist
+         FATAL("storage directory not found: %s", storage);
+      }   
+   } else if(writable) {
+      FATAL("'-w' not allowed when running out of storage");
```

Comment:
Allow this with `--unsafe`? Could simplify the test suite by taking advantage of this capability, I suspect.